File: lib/receipt.rb
require 'bigdecimal'
require_relative 'tax_calculator'

class Receipt
    attr_reader :items, :total_sales_tax, :total_amount

    def initialize(items)
        @items = items
        @total_sales_tax = BigDecimal('0')
        @total_amount = BigDecimal('0')
    end

    def generate
        items.each do |item|
            tax = TaxCalculator.get_tax(item)
            total_price = format('%.2f', (item.price + tax) * item.quantity)

            puts "#{item.quantity} #{item.name}: #{total_price}"

            @total_sales_tax += tax * item.quantity
            @total_amount += (item.price + tax) * item.quantity
        end
        puts "Sales Taxes: #{format('%.2f', total_sales_tax)}"
        puts "Total: #{format('%.2f', total_amount)}"
    end
end

File: lib/receipt.py
from decimal import Decimal

from lib.tax_calculator import get_tax


class Receipt:
    def __init__(self, items):
        self.items = items
        self.total_sales_tax = Decimal('0')
        self.total_amount = Decimal('0')

    def generate(self):
        for item in self.items:
            tax = get_tax(item)
            total_price = f"{(item.price + tax) * item.quantity:.2f}"
            print(f"{item.quantity} {item.name}: {total_price}")
            self.total_sales_tax += tax * item.quantity
            self.total_amount += (item.price + tax) * item.quantity
        print(f"Sales Taxes: {self.total_sales_tax:.2f}")
        print(f"Total: {self.total_amount:.2f}")
